### PR TITLE
OTP27 support; Fix handling of negative zero in OTP27+

### PR DESCRIPTION
**Pull request by v0idpwn**
Adds CI and fixes tests that fail on OTP27. 
Fixes dependency (ssl_verify_fun) that would cause CI to fail on Elixir 1.15+.
Fixes warning that would cause CI to fail on OTP25 + Elixir 1.16 (closing #369).

Took the opportunity to fix negative zero handling:

According to the official documentation in
https://protobuf.dev/programming-guides/proto3/#default as of 09/05/24,
if a float or double value is set to +0 it will not be serialized, but
if it is set to -0 it should be serialized.

Currently there are no conformance tests for this behaviour.

### CODE REVIEW

File: test/protobuf/dsl_test.exs [hunk 1]
```diff
@@ -165,18 +165,20 @@ defmodule Protobuf.DSLTest do
...
              "B" => :B,
              "C" => :C,
              "UNKNOWN" => :UNKNOWN,
              "D" => :D,
              "E" => :E
-           }
+           } = TestMsg.EnumFoo.__reverse_mapping__()
+
+    # Varies with erlang version
+    assert TestMsg.EnumFoo.__reverse_mapping__()[4] in [:C, :D, :E]
```
Comment (v0idpwn) [inline test/protobuf/dsl_test.exs:181]: This enum is aliased, hence it's correct to decode to any of the aliases, according to documentation:
>Though all alias values are valid during deserialization, the first value is always used when serializing.

(https://protobuf.dev/programming-guides/proto3/)

File: test/protobuf/json/decode_test.exs [hunk 2]
```diff
@@ -341,8 +341,8 @@ defmodule Protobuf.JSON.DecodeTest do
 
   describe "enums" do
     test "known integer value is valid" do
-      data = %{"j" => 4}
-      decoded = %Foo{j: :E}
+      data = %{"j" => 1}
+      decoded = %Foo{j: :A}
```
Comment (v0idpwn) [inline test/protobuf/json/decode_test.exs:345]: Here I just avoided using an aliased enum

File: test/protobuf/encoder_test.exs [hunk 3]
```diff
@@ -171,6 +171,17 @@ defmodule Protobuf.EncoderTest do
     assert Encoder.encode(msg) == <<>>
   end
 
+  if :erlang.system_info(:otp_release) < ~c"27" do
```
Comment (whatyouhide) [inline test/protobuf/encoder_test.exs:174]:
```suggestion
  if System.otp_version(:otp_release) < "27" do
```

File: test/protobuf/json/encode_test.exs [hunk 4]
```diff
@@ -98,6 +98,14 @@ defmodule Protobuf.JSON.EncodeTest do
     assert encode(message) == %{"float" => "-Infinity", "double" => "-Infinity"}
   end
 
+  if :erlang.system_info(:otp_release) < ~c"27" do
```
Comment (whatyouhide) [inline test/protobuf/json/encode_test.exs:101]:
```suggestion
  if System.otp_release(:otp_release) < "27" do
```

**Comment by v0idpwn** [comment]
Looks like erleft/setup-beam is too strict and doesn't allow me to use Elixir 1.16.2 with the rc OTP...

**Comment by whatyouhide** [comment]
Yeah we'll need to wait for OTP 27 to be released. Should be *real* soon.

**Comment by v0idpwn** [comment]
Should we maybe drop the CI part (can add it in another PR), or just hold this until its released? :)

**Comment by whatyouhide** [comment]
@v0idpwn can you remove the CI part so that we can merge this, and open an issue to remind us to bump CI when 27 is released?

**Comment by v0idpwn** [comment]
Rebased and changed to add Elixir 1.16.2 with OTP 25 instead of with OTP 27.

**Review by ericmj — APPROVED** [review]

**Comment by ericmj** [comment]
There's a CI failure, other than it looks good.

**Comment by whatyouhide** [comment]
Thanks @v0idpwn 🙏

**Comment by zhihuizhang17** [comment]
@whatyouhide, request a new version release in` hex.pm` pls.

**Pull request merged**